server/service/service_osquery.go

```diff
@@ -703,9 +725,21 @@ func (svc *Service) SubmitDistributedQueryResults(
 		status, ok := statuses[query]
 		failed := ok && status != fleet.StatusOK
 		switch {
+		// Details is a mixture of things that belong in the hosts table and things that don't and could be separated:
+		// users, software, pack stats
```

Comment:
+1, or maybe we could have `DetailQuery` define:
- `HostIngestFunc // udpdates host table`
- `HostExtraIngestFunc // updates those "secondary/extra" tables like host_software, software, etc.`.
(I think this is aligned with your idea but using more explicit host-table-vs.-no-host-table naming.)

Reply:
You mean as a naming change or as a deeper change in how things are structured?